Still two PEP8 issues:

```
./hr_employee_firstname/hr.py:36:70: E502 the backslash is redundant between brackets
./hr_employee_firstname/hr.py:37:38: F821 undefined name 'pg_varchar'
```

--- a/hr_employee_exemption/models/hr_employee.py
+++ b/hr_employee_exemption/models/hr_employee.py
@@ -45,7 +45,7 @@ class HrEmployee(orm.Model):
         if isinstance(ids, (int, long)):
             ids = [ids]
 
-        assert(len(ids), 1)
+        assert len(ids) == 1, 'must be called with a single employee'
 
         employee = self.browse(cr, uid, ids[0], context=context)
 

--- a/hr_employee_exemption/models/hr_salary_rule.py
+++ b/hr_employee_exemption/models/hr_salary_rule.py
@@ -46,7 +46,7 @@ class HrSalaryRule(orm.Model):
         if isinstance(ids, (int, long)):
             ids = [ids]
 
-        assert(len(ids), 1)
+        assert len(ids) == 1, 'must be called with a single employee'
 
         rule = self.browse(cr, uid, ids[0], context=context)
 

--- a/hr_employee_firstname/hr.py
+++ b/hr_employee_firstname/hr.py
@@ -26,17 +26,20 @@ from openerp.osv import orm, fields
 class hr_employee(orm.Model):
     _inherit = 'hr.employee'
 
-    def init(self, cursor):
-        cursor.execute('''\
-SELECT id
-FROM hr_employee
-WHERE lastname IS NOT NULL
-LIMIT 1''')
-        if not cursor.fetchone():
-            cursor.execute('''\
-UPDATE hr_employee
-SET lastname = name_related
-WHERE name_related IS NOT NULL''')
+    def _auto_init(self, cr, context=None):
+        """pre-create and fill column lastname so that the constraint
+        setting will not fail"""
+        self._field_create(cr, context=context)
+        column_data = self._select_column_data(cr)
+        if 'lastname' not in column_data:
+            field = self._columns['lastname']
+            cr.execute('ALTER TABLE "%s" ADD COLUMN "lastname" %s' %
+                       (self._table,
+                        orm.pg_varchar(field.size)))
+            cr.execute('UPDATE hr_employee '
+                       'SET lastname = name_related '
+                       'WHERE name_related IS NOT NULL')
+        return super(hr_employee, self)._auto_init(cr, context=context)
 
     def create(self, cursor, uid, vals, context=None):
         firstname = vals.get('firstname')

--- a/hr_expense_sequence/hr_expense_sequence.py
+++ b/hr_expense_sequence/hr_expense_sequence.py
@@ -28,6 +28,22 @@ class hr_expense_expense(orm.Model):
     _inherit = 'hr.expense.expense'
     _order = 'name desc'
 
+    def _auto_init(self, cr, context=None):
+        """pre-create and fill column description with the column name
+        so that the required constraint setting will not fail
+        """
+        self._field_create(cr, context=context)
+        column_data = self._select_column_data(cr)
+        if 'description' not in column_data:
+            field = self._columns['description']
+            cr.execute('ALTER TABLE "%s" ADD COLUMN "description" %s' %
+                       (self._table,
+                        orm.pg_varchar(field.size)))
+            cr.execute('UPDATE hr_expense_expense '
+                       'SET description = name '
+                       'WHERE name IS NOT NULL')
+        return super(hr_expense_expense, self)._auto_init(cr, context=context)
+
     _columns = {
         # Move the description from the 'name' field to 'description' field
         # In the 'name' field, we now store the number/sequence

--- a/hr_leave_accruals/res_company.py
+++ b/hr_leave_accruals/res_company.py
@@ -32,6 +32,6 @@ class res_company(orm.Model):
         ),
     }
 
-    _default = {
+    _defaults = {
         'holidays_hours_per_day': 8.0,
     }

--- a/hr_payroll_period/hr_payroll_period.py
+++ b/hr_payroll_period/hr_payroll_period.py
@@ -865,6 +865,21 @@ class hr_holidays_status(orm.Model):
     _name = 'hr.holidays.status'
     _inherit = 'hr.holidays.status'
 
+    def _auto_init(self, cr, context=None):
+        """pre-create and fill column code so that the constraint
+        setting will not fail"""
+        self._field_create(cr, context=context)
+        column_data = self._select_column_data(cr)
+        if 'code' not in column_data:
+            field = self._columns['code']
+            cr.execute('ALTER TABLE "%s" ADD COLUMN "code" %s' %
+                       (self._table,
+                        orm.pg_varchar(field.size)))
+            cr.execute('UPDATE hr_holidays_status '
+                       'SET code = substring(name from 1 for 16) '
+                       'WHERE code IS NULL')
+        return super(hr_holidays_status, self)._auto_init(cr, context=context)
+
     _columns = {
         'code': fields.char('Code', size=16, required=True),
     }

